packages/create-plugin/templates/datasource/src/datasource.ts

```diff
@@ -43,11 +49,46 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
     return { data };
   }
 
+  async request(url: string, params?: string) {
+    const response = getBackendSrv().fetch<DataSourceResponse>({
+      url: `${this.baseUrl}${url}${params?.length ? `?${params}` : ''}`,
+    });
+    return lastValueFrom(response);
+  }
+
+  /**
+   * Checks whether we can connect to the API.
+   */
   async testDatasource() {
-    // Implement a health check for your data source.
-    return {
-      status: 'success',
-      message: 'Success',
-    };
+    const defaultErrorMessage = 'Cannot connect to API';
+
+    try {
+      const response = await this.request('/health');
+      if (response.status === 200) {
+        return {
+          status: 'success',
+          message: 'Success',
+        };
+      } else {
+        return {
+          status: 'error',
+          message: response.statusText ? response.statusText : defaultErrorMessage,
+        };
+      }
+    } catch (err) {
+      let message = '';
+      if (_.isString(err)) {
```

Comment:
Hmm... lodash feels a bit overkill for checking primitives. Is this how we do it in examples repo? frontend platform has been trying to remove/discourage usage of lodash in core for quite some time now.

Reply:
Copied from the examples repo. Yeah I don't like this either. Will try to fix here and update examples repo too